skel/daemons/configuration.go

```diff
@@ -0,0 +1,373 @@
+				if walletKey == "partial" {
+					for _, w := range walletsGroup {
+						res, err := getWallet(peatioClient, w)
+						if err != nil {
+							// For example, If there is difference in second slice
+							// It means that second slice is not fully included on first slice
+							// []string{"eth", "link", "usdt"} VS []string{"eth", "link"}
+							// res: []string{"usdt"}, []string{}
+							_, secondDiff := funk.Difference(res.Currencies, currencyGroup)
+							if len(secondDiff.([]string)) > 0 {
+								mergedCurrencies := append(res.Currencies, currencyGroup...)
+								updateWallet(peatioClient, w, funk.UniqString(mergedCurrencies))
+							}
+						}
+					}
+				} else if walletKey == "none" {
+					createDepositAndHotWallet(peatioClient, currencyGroup, opendaxAddr)
+				}
+			}
+		}
+	}
+}
+
+// Method divideCurrenciesIntoGroups divides currencies into groups
+// In first place system always puts parent ID
+// For example {"eth": ["eth", "usdt", "link"]}
+func divideCurrenciesIntoGroups(currencies []CurrencyResponse) map[string][]string {
+	res := make(map[string][]string)
+	// Sort currencies with empty Parent ID first
+	// It means those currencies should have separate wallet
+	sort.SliceStable(currencies, func(i, j int) bool {
+		return currencies[i].ParentID < currencies[j].ParentID
+	})
+
+	for _, currency := range currencies {
+		if currency.Type == "coin" {
+			if currency.ParentID == "" {
+				res[currency.ID] = append(res[currency.ID], currency.ID)
+			} else {
+				res[currency.ParentID] = append(res[currency.ParentID], currency.ID)
+			}
+		}
+	}
+
+	return res
+}
+
+// Method createDepositAndHotWallet creates deposit and hot wallet
+func createDepositAndHotWallet(peatioClient *peatio.Client, currencies []string, opendaxAddr string) *mngapi.APIError {
+	// Create Deposit Wallet
+	depositWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Deposit Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "deposit",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
+
+	_, depositApiError := peatioClient.CreateWallet(depositWalletParams)
+	if depositApiError != nil {
+		log.Printf("ERROR: createWallets: Can't create deposit wallet. Error: %v. Errors: %v", depositApiError.Error, depositApiError.Errors)
+		return depositApiError
+	}
+
+	// Create Hot Wallet
+	hotWalletParams := peatio.CreateWalletParams{
+		BlockchainKey: "opendax-cloud",
+		Name:          fmt.Sprintf("%s Hot Wallet", strings.ToUpper(currencies[0])),
+		Kind:          "hot",
+		Gateway:       "opendax_cloud",
+		Address:       "address",
+		Currencies:    currencies,
+		Status:        "active",
+		Settings: peatio.Settings{
+			URI: fmt.Sprintf("%v/api/v2/opx/peatio", opendaxAddr),
+		},
+	}
```

Comment:
Let's use shared params for hot and deposit wallets(currencies, status, etc.), then we'd only need to override kind before the request